src/Program.cs

```diff
@@ -34,11 +35,30 @@ public static void Main(string[] args)
             }
 
             var visualizerDir = Environment.GetEnvironmentVariable("VISUALIZER_FILES") ?? ".";
-            var visualizer = new Visualizer(visualizerDir);
+            var chartLabels = GetChartLabels(tool, args);
+            var visualizer = new Visualizer(visualizerDir, chartLabels);
             var output = new FileStream(outputFileName, FileMode.OpenOrCreate, FileAccess.Write);
             visualizer.SaveChart(results, output);
         }
 
+        private static ChartLabels GetChartLabels(string tool, IReadOnlyList<string> args)
+        {
+            var url = new Uri(args[1], UriKind.Absolute);
+            switch (tool)
+            {
+                case "hammer":
+                {
+                    return new Hammer(new HttpClient(), url,  1).GetChartLabels();

```

Comment:
once the labels are public static, this part gets greatly simplified (no arg required, no tool instantiation, etc.)